format-avro/src/main/java/io/cdap/plugin/format/avro/AvroToStructuredTransformer.java

```diff
@@ -91,9 +95,76 @@ public Schema convertSchema(org.apache.avro.Schema schema) throws IOException {
     if (schemaCache.containsKey(hashCode)) {
       structuredSchema = schemaCache.get(hashCode);
     } else {
-      structuredSchema = Schema.parseJson(schema.toString());
+      String strSchema = schema.toString();
+      String jsonSchema = preprocessSchema(new Gson().fromJson(strSchema, JsonObject.class)).toString();
```

Comment:
Please don't create `Gson` each time, it's pretty expensive. Follow example of other places where we create it once for a class.

Reply:
Agree. `gson` should be a global instance.